modules/stochastic_tools/src/vectorpostprocessors/SamplerData.C

```diff
@@ -0,0 +1,56 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+// Stocastic Tools Includes
+#include "SamplerData.h"
+
+// MOOSE includes
+#include "Sampler.h"
+
+template <>
+InputParameters
+validParams<SamplerData>()
+{
+  InputParameters params = validParams<GeneralVectorPostprocessor>();
+  params.addClassDescription(
+      "Tool for extracting Sampler object data and storing in VectorPostprocessor vectors.");
+  params += validParams<SamplerInterface>();
+  params.addRequiredParam<SamplerName>("sampler",
+                                       "The sample from which to extract distribution data.");
+  return params;
+}
+
+SamplerData::SamplerData(const InputParameters & parameters)
+  : GeneralVectorPostprocessor(parameters), SamplerInterface(this), _sampler(getSampler("sampler"))
+{
+}
+
+void
+SamplerData::initialize()
+{
+  for (VectorPostprocessorValue * ptr : _sample_vectors)
```

Comment:
I am getting back on my soap box.

Why? VectorPostprocessorValue is already a typedef. Adding auto just masks that fact that that we are iterating over a vector, so then I have to go look at the header to see what type is underneath.

I just don't understand why we are going to auto so much. Sure I know we could name the iterator, like I did here, to indicate that the container is made of pointers, but that changes the code readability from using compile time errors on types to arbitrary variable names that will be easily wrong of overlooked.

We are using beginIndex because we can't be trusted to use the correct type, but we are going to rely on iterator names to make the code readable. This is all crazy to me.

I will get off the soap box now; I will change it to auto.